### PR TITLE
Updated Color method to return an error if the color name given is not valid, else return nil, 

### LINKED ISSUE #35: Color method isn't actually returning an error when it should

### PR DESCRIPTION
**Pull request by ClaytonNorthey92**
- added map that maps string-representation of color to it's foreground text color attribute, use this in the Color method so if the string passed in represents a valid color, the correct attribute is returned or else an error is returned
- test case TestColorError was added to test against both cases of valid and invalid color name passed into Color method
- fixes #35

### CODE REVIEW

File: spinner_test.go [hunk 1]
```diff
@@ -178,6 +178,23 @@ func TestBackspace(t *testing.T) {
...
 
+// TestColorError tests that if an invalid color string is passed to the Color
+// function, the invalid color error is returned
+func TestColorError(t *testing.T) {
+	s := New(CharSets[0], 100*time.Millisecond)
+
+	invalidColorName := "bluez"
+	validColorName := "green"
+
+	if s.Color(invalidColorName) != errInvalidColor {
```
Comment (ClaytonNorthey92) [inline spinner_test.go]: I don't like this...I should be checking for _nil_.

Comment (ClaytonNorthey92) [inline spinner_test.go]: I don't like this, I should be checking for _nil_ here, not simply the fact that it is not a specific error.

**Comment by briandowns** [comment]
Let me know when you're all done. :)

**Comment by ClaytonNorthey92** [comment]
hahaha I am done, sorry about that!

**Comment by briandowns** [comment]
No worries at all. Taking a look now.

File: spinner.go [hunk 2]
```diff
@@ -137,34 +148,14 @@ func (s *Spinner) Reverse() {
 
 // Color will set the struct field for the given color to be used
 func (s *Spinner) Color(c string) error {
+	var colorError error
```
Comment (briandowns) [inline spinner.go]: Is the additional error var necessary if we do something like below?

```go
func (s *Spinner) Color(c string) error {
    if !validColor(c) {
        return errInvalidColor
    }
    s..color = color.New(colorAttributeMap[c]).SprintFunc()
    s.Restart()
    return nil
}
```

Comment (ClaytonNorthey92) [inline spinner.go]: ah you're right it wouldn't be, I just like to have 1 return statement per method/function if I can help it, I can update it if you feel that way is better, both make sense to me

Comment (briandowns) [inline spinner.go]: yeah, update to something along those lines. Early returns are pretty common with Go.

Comment (ClaytonNorthey92) [inline spinner.go]: cool, on it

**Comment by briandowns** [comment]
LGTM. Thanks @ClaytonNorthey92 for the work on this!

**Comment by ClaytonNorthey92** [comment]
@briandowns no problem! Thank you!

**Pull request merged**